src/main/java/org/jvnet/hudson/tools/versionnumber/VersionNumberCommon.java

```diff
@@ -95,8 +110,18 @@ public static VersionNumberBuildInfo incBuild(Run build, Run prevBuild, boolean
                 buildsThisYear = 1;
             }
 
-            // increment total builds
-            buildsAllTime = info.getBuildsAllTime() + buildInc;
+            /*
+			 * If there's a valid override, we'll keep it, otherwise increment as normal.
+			 * 
+			 * TODO:  Not totally thrilled with this solution, but....
+			 * 
+			 * It's necessary to test this again here since we don't want to increment
+			 * the number if we've already resolved this from the beginning of the method.
+             */
```

Comment:
Hi Bagira,

Thanks for taking a look.  I'll see what it would take to manage all of the overrides in addition to just ``buildsAllTime``.

To answer your other question, I'm not thrilled with the approach either.  The issue is as follows:

```
        if (isValidOverrideBuildsAllTime(environmentVariables, overrideBuildsAllTime)) {
        	buildsAllTime = getOverrideBuildsAllTime(environmentVariables, overrideBuildsAllTime);
        }
        
        if (prevBuild != null) {
```

The issue is arising where we are checking for a ``prevBuild``.  If there's no ``prevBuild``, we were not entering the loop and not doing anything to allow us to override ``buildsAllTime``.  This basically meant that we couldn't override ``buildsAllTime`` on the very first build of a new project.  Since we need that may be a common use case of changing ``buildsAllTime`` (e.g. starting a new jenkins build for a pre-existing project where the version must be bumped), I need to figure it out here.

Then, we need to check it out again further down on 114.  This is because we need to know whether or not we successfully overrode the version.  This is to solve for the situation where someone passes an environment variable for ``buildsAllTime`` that doesn't resolve in the context of the current build.

I'm not super thrilled with the code quality either.  Let me take another pass and see about adding the other properties.  I was sort of trying to avoid larger changes, but given that we have decent tests, it's probably ok.

I'll make some adjustments and file another PR.